### ISSUE TITLE
Enum classes with companion object and without entries 

### ISSUE THREAD

**Issue by Dominaezzz** [square/kotlinpoet#621]
Instead of generating,
```
enum class RenderPassCreate(override val value: Int) {
    ;
    companion object
}
```
it generates,
```
enum class RenderPassCreate(override val value: Int) {
    companion object
}
```

It seems Kotlin requires the semi colon when the enum class has members.
`RenderPassCreate.kt: (7, 89): Expecting ';' after the last enum entry or '}' to close enum class body`

**Comment by Egorand** [comment]
Oh, that's an interesting edge case, thanks for filing! 

Looks like the semicolon is required every time the enum has anything besides constants. Would you be interested in submitting a fix? I also feel like this should be handled better by the compiler, will try to find/file the YouTrack issue.

**Comment by Egorand** [comment]
Closed with #622
